File: backend/Controllers/AdminController.cs
using Microsoft.AspNetCore.Mvc;
using JobPortal.API.Services;

namespace JobPortal.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class AdminController : ControllerBase
    {
        private readonly IAdminService _adminService;

        public AdminController(IAdminService adminService)
        {
            _adminService = adminService;
        }

        [HttpGet("pending-employers")]
        public async Task<IActionResult> GetPendingEmployers()
        {
            var result = await _adminService.GetPendingEmployersAsync();
            return Ok(result);
        }

        [HttpPost("approve-employer/{id}")]
        public async Task<IActionResult> ApproveEmployer(Guid id)
        {
            await _adminService.ApproveEmployerAsync(id);
            return Ok(new { Message = "Employer approved successfully" });
        }

        [HttpGet("pending-jobs")]
        public async Task<IActionResult> GetPendingJobs()
        {
            var result = await _adminService.GetPendingJobsAsync();
            return Ok(result);
        }

        [HttpPost("approve-job/{id}")]
        public async Task<IActionResult> ApproveJob(Guid id)
        {
            await _adminService.ApproveJobAsync(id);
            return Ok(new { Message = "Job published successfully" });
        }

        [HttpGet("metrics")]
        public async Task<IActionResult> GetMetrics()
        {
            var result = await _adminService.GetSystemMetricsAsync();
            return Ok(result);
        }
    }
}

File: backend/src/controllers/adminController.js
import { AdminService } from '../services/adminService.js';

export const adminController = {
  getPendingEmployers: (req, res) => {
    try {
      const pending = AdminService.getPendingEmployers();
      res.json(pending);
    } catch (err) {
      res.status(500).json({ message: err.message });
    }
  },

  approveEmployer: (req, res) => {
    try {
      const user = AdminService.approveEmployer(req.params.id);
      res.json({ message: 'Employer approved successfully', user });
    } catch (err) {
      res.status(404).json({ message: err.message });
    }
  },

  getPendingJobs: (req, res) => {
    try {
      const pending = AdminService.getPendingJobs();
      res.json(pending);
    } catch (err) {
      res.status(500).json({ message: err.message });
    }
  },

  approveJob: (req, res) => {
    try {
      const job = AdminService.approveJob(req.params.id);
      res.json({ message: 'Job published successfully', job });
    } catch (err) {
      res.status(404).json({ message: err.message });
    }
  },

  getMetrics: (req, res) => {
    try {
      const metrics = AdminService.getMetrics();
      res.json(metrics);
    } catch (err) {
      res.status(500).json({ message: err.message });
    }
  }
};
